rest-service/manager_rest/resource_manager.py

```diff
@@ -1398,6 +1398,20 @@ def get_deployment_parents_from_labels(labels):
                 parents.append(value)
         return parents
 
+    def get_deployment_object_types_from_labels(self, deployment, labels):
+        new_labels = self.get_labels_to_create(deployment, labels)
+        created_types = set()
+        delete_types = set()
+        for key, value in new_labels:
+            if key == 'csys-obj-type' and value:
+                created_types.add(value.lower())
+
+        new_labels_set = set(labels)
+        for label in deployment.labels:
+            if (label.key, label.value) not in new_labels_set:
+                delete_types.add(label.value.lower())
```

Comment:
This takes any label, and not only the ones with the key `csys-obj-type`. So basically also the label `a: environment`  will be counted.

Reply:
you are right, updated that and add check before add any deleted types